consume.go

```diff
@@ -65,6 +84,10 @@ func (cmd *consumeCmd) run(args []string) error {
 	if err != nil {
 		return fmt.Errorf("bad -valuecodec argument: %v", err)
 	}
+	if cmd.keyCodecType == "json" {
```

Comment:
I see why, but it's odd to check for this specific value - surely any invalid value could be handled in roughly this way?

Reply:
encoderForType will handle invalid values, but it will also accept `"json"`, which we don't want it to (because it leads to complications down the line).